enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,502 @@
+Specifically the [PlatformSpec](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L47)
+and [PlatformStatus](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L65) parts
+of the infrastructure resource.
+
+Given that PlatformSpec and PlatformStatus are defined as "discriminated unions" and are required to have the platform type encoded within
+"openshift/api" beforehand, it creates a significant footprint and plenty of effort from Red Hat engineers to create the initial technical enablement of
+a new cloud provider and mostly impossible without Red Hat engineering engagement.
+
+Since a lot of infrastructure related components (such as [CCCMO](https://github.com/openshift/cluster-cloud-controller-manager-operator), [Machine API operator](https://github.com/openshift/machine-api-operator), [Machine Config Operator](https://github.com/openshift/machine-config-operator), and so on)
+require information about, at least, the presence of an underlying cloud provider, the "None" platform does not fit well as a signal in such a situation.
+
+Having a special, built-in, and somewhat generic platform type that will signal about the presence of an underlying infrastructure platform without
+platform-specific details will help to reduce the number of changes across OCP repositories
+and simplify the initial integration work for non-Red Hat contributors.
+
+Examples of difficulties that are present today due to having a predefined list of platforms:
+- No defined mechanism to set `--cloud-provider=external` arg to kubelet/KCM/apiserver without merges and further revendoring of "openshift/api"
+- No way to extend machine-api and deliver a new provider without merges to "openshift/machine-api-operator" and "openshift/api" repos
+
+### Conjunction with Capabilities
+
+In the future, at some degree, [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+approach might help to solve the issue and give a room for supplementing *some* platform-dependent components, however some integral parts of
+OpenShift can not be disabled and still requires a signal about an underlying platform, for example, KCM and MCO, in respect of the enablement of
+[external cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
+
+One of the possible examples of the interaction of the capabilities with the "External" platform type would be the MachineAPI and the Machine Api Operator.
+At this point, the MachineAPI has no use in running if there is no machine controller (which is heavily cloud-provider dependent).
+"External" platform type will serve as a signal in the scenario when machine-api capability is enabled and the cluster deployer
```

Comment:
Suggestion to make the language a bit easier to follow:

"When the platform type is set to "External", and the machine-api capability is enabled, that will cause Machine API operator to deploy only generic cloud-independent controllers (such as machine-healthcheck, machineset, and node-link controllers). The platform-specific components would be deployed through a separate mechanism. Such behaviour will simplify initial cloud-platform enablement and will reduce the necessity of reverse-engineering and replicating work that was already done by Red Hat engineers."

Reply:
Agree, thanks!